nexus/tests/integration_tests/pantry.rs

```diff
@@ -0,0 +1,821 @@
+
+    DiskTest::new(&cptestctx).await;
+    create_org_and_project(client).await;
+
+    create_disk_with_state_importing_blocks(client).await;
+
+    // Validate disk is in state ImportReady
+    validate_disk_state(client, DiskState::ImportReady).await;
+
+    // bulk write start
+    bulk_write_start(client, StatusCode::NO_CONTENT).await;
+
+    // Validate disk is in state ImportingFromBulkWrites
+    validate_disk_state(client, DiskState::ImportingFromBulkWrites).await;
+
+    let bulk_write_url = format!(
+        "/v1/disks/{}/bulk-write?project={}&organization={}",
+        DISK_NAME, PROJECT_NAME, ORG_NAME,
+    );
+
+    NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, &bulk_write_url)
+            .body(Some(&params::ImportBlocksBulkWrite {
+                offset: 0,
+                base64_encoded_data: "this is not base64!".to_string(),
+            }))
+            .expect_status(Some(StatusCode::BAD_REQUEST)),
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .unwrap();
+}
+
+// Test that users can stop and start importing using bulk writes multiple times
+#[nexus_test]
+async fn test_can_stop_start_import_from_bulk_write(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let client = &cptestctx.external_client;
+
+    DiskTest::new(&cptestctx).await;
+    create_org_and_project(client).await;
+
+    create_disk_with_state_importing_blocks(client).await;
+
+    // Validate disk is in state ImportReady
+    validate_disk_state(client, DiskState::ImportReady).await;
+
+    for _ in 0..16 {
+        bulk_write_start(client, StatusCode::NO_CONTENT).await;
+        validate_disk_state(client, DiskState::ImportingFromBulkWrites).await;
+
+        bulk_write_stop(client).await;
+        validate_disk_state(client, DiskState::ImportReady).await;
+    }
+
+    finalize_import(client, StatusCode::NO_CONTENT).await;
+}
+
+// Test that users cannot start a bulk write when a disk is attached to an
+// instance
+#[nexus_test]
+async fn test_cannot_bulk_write_start_attached_disk(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let client = &cptestctx.external_client;
+    let nexus = &cptestctx.server.apictx().nexus;
+
+    DiskTest::new(&cptestctx).await;
+    create_org_and_project(client).await;
+
+    create_disk_with_state_importing_blocks(client).await;
+
+    // bulk write bytes in and finalize
+    bulk_write_start(client, StatusCode::NO_CONTENT).await;
+    bulk_write_bytes(client).await;
+    bulk_write_stop(client).await;
+    finalize_import(client, StatusCode::NO_CONTENT).await;
+
+    // Validate disk is in state detached
+    validate_disk_state(client, DiskState::Detached).await;
+
+    // Create an instance to attach the disk.
+    create_instance_and_attach_disk(client, nexus, StatusCode::ACCEPTED).await;
+
+    // Validate that a user cannot start a bulk write
+    bulk_write_start(client, StatusCode::BAD_REQUEST).await;
+}
```

Comment:
Throw a `bulk_write_bytes` and `bulk_write_stop` in here and cover those cases as well.

Reply:
right, done in 107bb6fe